### ISSUE TITLE
Review the accessibility of our icons

### ISSUE THREAD

**Issue by GomezIvann** [dxc-technology/halstack-react#1449]
**Is your feature request related to a problem? Please describe.**
Currently, our icons (not the custom ones) are not accessible. For example, the expand and contract icons from the Select, Dropdown and Sidenav are not properly self-explained for a blind user who uses screen readers. 

![image](https://user-images.githubusercontent.com/44321109/214296003-d618e543-2b2c-4822-9b66-68481b7fb45c.png)

We need to review every icon of our components that cannot be customised and add the necessary attributes to make it accessible.

**Describe the solution you'd like**
- Start using the [`title` and `desc`](https://www.w3.org/TR/SVG11/struct.html#DescriptionAndTitleElements) attributes when needed.
- Reference the title with the svg by using `aria-labelledby`.
- Use `role='img'` when needed.

**Additional context**
- https://dequeuniversity.com/rules/axe/4.2/svg-img-alt
- https://css-tricks.com/accessible-svgs/
- https://codepen.io/Nice2MeatU/pen/dqmypX
- https://a11y-101.com/development/svg

**Comment by Mil4n0r** [comment]
I am closing this issue as this will be covered inside #1855
